### PR TITLE
v2 Menus: Implement typeahead focus

### PR DESCRIPTION
**Pull request by siddharthkp**
- ~Waiting on https://github.com/primer/behaviors/pull/59~
- Merge after https://github.com/primer/react/pull/1877

---

- Fixes: https://github.com/github/primer/issues/588
- [Storybook](https://primer-react-mmsijbnzz-primer.vercel.app/react/storybook?path=/story/composite-components-actionmenu2-fixtures--typeahead-test)
- [Test fixtures](https://github.com/primer/react/pull/1834/files#diff-57733dc833f4808c6c4f32601b1ca72133ad827745ee0292e184da0853344e1bR34)

--- 

### Looking for feedback on:

1. ~Is this correct behavior - Typeahead does not activate automatically when you open with mouse click. But, if you press the arrow keys after opening with mouse, typeahead becomes activated.~
2. Is the typeahead timeout 1000ms appropriate? The timeout is necessary so that you can type a longer string to match options. At the same, if you have found the matching element quickly before the timeout runs wouldn't select the option, making it feel unresponsive. [See storybook setup](https://primer-react-mmsijbnzz-primer.vercel.app/react/storybook?path=/story/composite-components-actionmenu2-fixtures--typeahead-test) to play around with it.

--- 

### Screenshots:

Before:
Pressing alphabet keys does nothing

After:

Pressing alphabet keys changes focus the matching element

---

100% test coverage yo!

image

### CODE REVIEW

File: src/ActionMenu2.tsx
Comment (siddharthkp) [inline src/ActionMenu2.tsx:100]: Changed both of these hooks to accept a ref if it's passed (or create a new one if it's not passed)

**Comment by colebemis** [comment]
> Is this correct behavior - Typeahead does not activate automatically when you open with mouse click. But, if you press the arrow keys after opening with mouse, typeahead becomes activated.

I'm also curious about this ☝️  cc @alliethu 

It seems a little strange that typeahead doesn't work when activated with a mouse, but I don't have a strong opinion.

**Review by colebemis — APPROVED** [review]
Nice work @siddharthkp! Looks good to me 👍  I would like an answer to https://github.com/primer/react/pull/1834#issuecomment-1033034968 from someone with more accessibility experience before we merge this, but feel free to merge if this is urgent.

Side note, this would be fun to implement as a [state machine](https://twitter.com/statelyai/status/1491017367815884800?s=20&t=s-ERtpiik9yaf--5dXAzPw)!

**Comment by siddharthkp** [comment]
Update: After https://github.com/primer/react/pull/1877, typeahead is active even when you open the menu with mouse click ✌️

**Pull request merged**
